packages/eslint-plugin/src/rules/no-void-expression.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  AST_NODE_TYPES,
+  TSESLint,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+import * as util from '../util';
+import {
+  isClosingParenToken,
+  isOpeningParenToken,
+  isParenthesized,
+} from '../util';
+
+export type Options = [
+  {
+    ignoreArrowShorthand?: boolean;
+    ignoreVoidOperator?: boolean;
+  },
+];
+
+export type MessageId =
+  | 'invalidVoidExpr'
+  | 'invalidVoidExprWrapVoid'
+  | 'invalidVoidExprArrow'
+  | 'invalidVoidExprArrowWrapVoid'
+  | 'invalidVoidExprReturn'
+  | 'invalidVoidExprReturnLast'
+  | 'invalidVoidExprReturnWrapVoid'
+  | 'voidExprWrapVoid';
+
+export default util.createRule<Options, MessageId>({
+  name: 'no-void-expression',
+  meta: {
+    docs: {
+      description:
+        'Requires expressions of type void to appear in statement position',
+      category: 'Best Practices',
+      recommended: false,
+      requiresTypeChecking: true,
+    },
+    messages: {
+      invalidVoidExpr:
+        'Placing a void expression inside another expression is forbidden. ' +
+        "Move it to it's own statement instead.",
+      invalidVoidExprWrapVoid:
+        'Void expressions used inside another expression ' +
+        'must be marked explicitly with the `void` operator.',
```

Comment:
I didn't want to make this message too long but for this case it would actually make sense, because it's probably an actual error.

I changed it to

> Void expressions used inside another expression must be moved to its own statement or marked explicitly with the `void` operator.

Also, I noticed it should be "its" not "it's". Or maybe "their". I'm not a native english speaker so I'm not sure.

Reply:
`it's` is correct, `their` would probably fit for a group of statements but since it Is applied for each one `it's` is fine.

Reply:
Are you sure?

https://www.grammarly.com/blog/its-vs-its/

Reply:
nope, I'm wrong. English is hard. 😝 possessive is `its`